paderbox/utils/nested.py

```diff
@@ -160,7 +160,7 @@ def nested_merge(default_dict, *update_dicts, allow_update=True, inplace=False):
         if inplace:
             return default_dict
         else:
-            return default_dict.__class__(default_dict)
+            return copy.deepcopy(default_dict)
```

Comment:
Why do you change to `deepcopy`? `copy` does exactly what is needed here:
 - Only copy the outer dict, not the values (i.e. avoid a copy of the "leafs")
 - The recursive call takes care to handle the inner dicts.

Reply:
The non-shortcut code path implicitly performs a deep copy, so the shortcut (no update) code path should do the same. There is no recursive call here.

Reply:
Ok, it only performs a deep copy for the mapping types and not for any other types

Reply:
Oh, I was not carefully checking the code. Here is nothing recursive.
Nonetheless, we should not copy the leafs.
The leafs could be large and then the copy is slow and increases the memory consumption.

I would keep the old behavior to have a trade off between speed and what the user expects.